site/content/features/stringtemplates.md

```diff
@@ -0,0 +1,466 @@
+
+The name `StringTemplate` is perhaps unfortunate.  *Any* of the four possible template argument types is turned into a `StringTemplate` instance. `TemplateArguments` might have been a better choice for the class name.
+
+A `StringTemplate` instance stores this information:
+
+* The *fragments*; that is, the strings between the embedded expressions
+* The values of the embedded expressions
+
+The `fragments` and `values` methods yield lists of fragments and values.
+
+```
+String name = "Fred";
+int age = 42;
+StringTemplate template = RAW."Hello, \{name}! Next year, you'll be \{age + 1}.";
+template.fragments() // ["Hello, ", "! Next year, you'll be ", "."]
+template.values() // ["Fred", 43]
+```
+
+Note that there is one more fragment than there are values.
+
+The static `interpolate` method merges a list of fragments and values. As you will see in the next section, this is useful for implementing your  own template processors.
+
+I don't think that many programmers will create `StringTemplate` instances, but there are three ways to do so:
+
+* With the `RAW` processor:
+          ```
+RAW."Hello, \{name}!"
+```
+
+
+* With the static `of` methods:
+          ```
+StringTemplate.of(List("Hello, ", "!"), List(name))
+StringTemplate.of("Hello, Sailor!")
+```
+
+
+* With the static `combine` methods:
+          ```
+StringTemplate.combine(RAW."Hello, \{name}! ", RAW."Next year, you'll be \{age + 1}.")
+```
+
+
+
+There are two `combine` methods, with parameter types `StringTemplate...` and `List<StringTemplate>`.
+
+Finally, the `process` method lets you apply a processor, just in case you prefer
+
+```
+template.process(processor)
+```
+
+over
+
+```
+processor.process(template)
+```
+
+This sandbox shows the API in action, even though you probably only need to worry about the first two methods. 
+
+{{< sandbox version=java19 preview="true" mainclass="StringTemplateDemo" >}}{{< sandboxsource "StringTemplateDemo.java" >}}
+import static java.lang.StringTemplate.RAW;
+import java.util.List;
+
+public class StringTemplateDemo {
+   public static void main(String[] args) {
+      String name = "Fred";
+      int age = 42;
+      StringTemplate template = RAW."Hello, \{name}! Next year, you'll be \{age + 1}.";
+      System.out.println(template.fragments());
+      System.out.println(template.values());
+
+      // Two interpolate methods: one static, one instance
+      System.out.println(StringTemplate.interpolate(
+         template.fragments().stream().map(String::toUpperCase).toList(), template.values()));
+      System.out.println(StringTemplate.of(
+         template.fragments().stream().map(String::toUpperCase).toList(),
+            template.values()).interpolate());
+
+      // Demo of the of and combine methods.
+      // For each entry in System.getProperties(), one StringTemplate is created.
+      // They are all combined and then processed with the STR processor
+      template = StringTemplate.combine(
```

Comment:
Side note: I was trying to use `Stream.reduce()` but then realized that I miss a constant for an *empty* `StringTemplate`. So I had to write:

    .reduce(StringTemplate.of(List.of(""), List.of()), StringTemplate::combine);

Also somehow the rule that there has to be one more fragment looks a bit questionable here.